File: 850879_c09_download/Ch09/Code/Ch09_ProSSIS/SSISUtilityLib_CSharp/SSISUtilityLib_CSharp/DataUtilities.cs
using System;
using System.Text;
using System.Text.RegularExpressions;
using System.Data;
using System.Data.SqlClient;

namespace SSISUtilityLib_CSharp
{
    public static class DataUtilities
    {
        public static bool isValidUSPostalCode(string PostalCode)
        {
            return Regex.IsMatch(PostalCode, "^[0-9]{5}(-[0-9]{4})?$");
        }
    }


}

File: 850879_c09_download/Ch09/Code/Ch09_ProSSIS/SSISUtilityLib_VB/SSISUtilityLib_VB/DataUtilities.vb
Imports System.Text.RegularExpressions

Public Class DataUtilities
    Public Shared Function isValidUSPostalCode(ByVal PostalCode As String) As Boolean
        isValidUSPostalCode = Regex.IsMatch(PostalCode, "^[0-9]{5}(-[0-9]{4})?$")
    End Function
End Class
